pkg/controllers/user/rbac/namespace_handler.go

```diff
@@ -118,6 +120,39 @@ func (n *nsLifecycle) ensurePRTBAddToNamespace(ns *v1.Namespace) (bool, error) {
 			return false, errors.Wrapf(err, "couldn't ensure binding %v in %v", prtb.Name, ns.Name)
 		}
 	}
+
+	var namespace string
+	if parts := strings.SplitN(projectID, ":", 2); len(parts) == 2 && len(parts[1]) > 0 {
+		namespace = parts[1]
+	} else {
+		return hasPRTBs, nil
+	}
+
+	rbs, err := n.m.rbLister.List(ns.Name, labels.Set(map[string]string{}).AsSelector())
+	if err != nil {
+		return false, errors.Wrapf(err, "couldn't list role bindings in %s", ns.Name)
+	}
+	client := n.m.workload.RBAC.RoleBindings(ns.Name).ObjectClient()
+
+	for _, rb := range rbs {
+		if uid := convert.ToString(rb.Labels[rtbOwnerAnn]); uid != "" {
+			prtbs, err := n.m.prtbIndexer.ByIndex(prtbByUIDIndex, uid)
+			if err != nil {
+				continue
+			}
+			for _, prtb := range prtbs {
+				if prtb, ok := prtb.(*v3.ProjectRoleTemplateBinding); ok {
+					if prtb.Namespace != namespace {
+						logrus.Infof("Deleting role binding %s in %s", rb.Name, ns.Name)
+						if err := client.Delete(rb.Name, &metav1.DeleteOptions{}); err != nil {
```

Comment:
if the the error is a not found error, we should not return it, just continue on